src/views/Layout.jsx

```diff
@@ -10,17 +15,83 @@ import './Layout.css';
  * defined in `App.jsx`.
  */
 
-export function Layout() {
+export function Layout({ data, listToken }) {
+	const [showListToken, setShowListToken] = useState(false);
+	const [copiedToken, setCopiedToken] = useState(false);
+
+	function toggleListToken() {
+		setShowListToken((prevState) => !prevState);
+	}
+
+	function copyTokenToClipboard() {
+		window.navigator.clipboard.writeText(listToken);
+		setCopiedToken(true);
+
+		setTimeout(() => {
+			setShowListToken(false);
+			setCopiedToken(false);
+		}, 2000);
+	}
+
 	return (
 		<>
-			<div className="Layout">
-				<header className="Layout-header">
-					<h1>Smart shopping list</h1>
-				</header>
+			<div className="h-screen md:max-w-md md:mx-auto">
+				{listToken && (
```

Comment:
@neshacascia changed so that it doesn't appear on home page.

```suggestion
				{currentPath !== "/" && (
```

Reply:
I'm going to leave this as is since checking if `listToken` is `true` ensures that it will never render on the home page. The only time users will be on the homepage is when they have yet to create a `listToken` or join an existing list.